### PR TITLE
Refactor sanitizeHTML method to support attributes

### PR DESCRIPTION
**Pull request by axlwaii**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+      // If the tag is allowed, we'll retain only allowed attributes.
+      if (closing) {
+        // If it's a closing tag, simply return it as is.
+        return `</${tagName}>`;
+      }
+      // Otherwise, reconstruct the opening tag with only allowed attributes.
+      let allowedAttrs = attrs.split(/\s+/)
+          .filter(attr => allowedAttributes.includes(attr.split('=')[0]))
+          .join(' ');
+      return `<${tagName}${allowedAttrs ? ' ' + allowedAttrs : ''}>`;
```
Comment (LostCrew) [inline src/utils.js:31]:
```suggestion
      return `<${tagName}${allowedAttrs ? ` ${allowedAttrs}` : ''}>`;
```

File: src/utils.js [hunk 2]
```diff
@@ -1,37 +1,36 @@
...
-  return htmlWithoutAttributes.replace(regExp, '');
+  return htmlString.replace(/<(\/*)(\w+)([^>]*)>/g, (match, closing, tagName, attrs) => {
+    if (allowedTags.includes(tagName)) {
+      // If the tag is allowed, we'll retain only allowed attributes.
+      if (closing) {
+        // If it's a closing tag, simply return it as is.
+        return `</${tagName}>`;
+      }
+      // Otherwise, reconstruct the opening tag with only allowed attributes.
+      let allowedAttrs = attrs.split(/\s+/)
```
Comment (LostCrew) [inline src/utils.js:28]: Never re-assigned, so it should be `const`. Didn't the linter automatically change it?

Comment (axlwaii) [reply src/utils.js]: Good catch. The linter did not fail - I'll look into this in a follow up.

File: src/utils.js [hunk 3]
```diff
@@ -1,37 +1,36 @@
...
-    // '\/?' Matches / zero to one time for the closing tag
-    // '\s*?' Optional match of any white space charater after optional /
-    // '(${allowedTags.join('|')})>' matching group of the allowed tags
-    // ')' close the matching group of negative lookup
-    // '\w*[^<>]*' matches any word that isn't in the excluded group
-    // '>' Match closing tagq
-    `<(?!\\s*\\/?\\s*(${allowedTagsWhiteSpaced.join('|')})>)\\w*[^<>]*>` :
-    // Strips all tags
-    '<(\\/?\\w*)\\w*[^<>]*>';
+  if(allowedTags.length === 0) {
```
[the remark this answers is not in the document]
Comment (axlwaii) [reply src/utils.js]: Ran the linter manually and fixed the linting issues.

**Review by LostCrew — APPROVED** [review]
💯 Let's cut a 3.0.0 after merging this

**Pull request merged**